### ISSUE TITLE
[1.12.1] node-agent metrics missing

### ISSUE THREAD

**Issue by HaveFun83** [vmware-tanzu/velero#7014]
**What steps did you take and what happened:**

Since rollout of velero 1.12.0 the node-agent prometheus metrics are missing 
```bash
kubectl port-forward -n velero node-agent-cwhjc 8085
curl localhost:8085/metrics
curl: (52) Empty reply from server
```
Prometheus shows an "connect: connection refused" towards the node-agent pods

**What did you expect to happen:**

node-agent prometheus metrics are available on the desired port

**Anything else you would like to add:**

**Environment:**

- Velero version (use `velero version`): v1.12.0 / v1.12.1
- Velero features (use `velero client config get features`):  -
- Kubernetes version (use `kubectl version`): v1.27.4
- Kubernetes installer & version: Kubeadm
- Cloud provider or hardware configuration: hardware
- OS (e.g. from `/etc/os-release`): Fedora CoreOS 38.20231002.3.1

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by HaveFun83** [comment]
Found the issue 
since release 1.12.0 the node-agent metric port changed  to 8080
log from a node-agent
```
time="2023-10-25T10:16:27Z" level=info msg="Setting log-level to INFO"
time="2023-10-25T10:16:27Z" level=info msg="Starting Velero node-agent server v1.12.1 (5c4fdfe147357ec7b908339f4516cd96d6b97c61-dirty)" logSource="pkg/cmd/cli/nodeagent/server.go:103"
1.698228987275461e+09	INFO	controller-runtime.metrics	Metrics server is starting to listen	{"addr": ":8080"}
time="2023-10-25T10:16:27Z" level=info msg="Starting metric server for node agent at address []" logSource="pkg/cmd/cli/nodeagent/server.go:229"
time="2023-10-25T10:16:39Z" level=info msg="Starting controllers" logSource="pkg/cmd/cli/nodeagent/server.go:245"
time="2023-10-25T10:16:39Z" level=info msg="Controllers starting..." logSource="pkg/cmd/cli/nodeagent/server.go:287"
1.698228999510614e+09	INFO	Starting server	{"path": "/metrics", "kind": "metrics", "addr": "[::]:8080"}
```
but the pod spec point to 8085
```
    ports:
    - containerPort: 8085
      name: http-monitoring
      protocol: TCP
```

**Comment by HaveFun83** [comment]
maybe related to https://github.com/vmware-tanzu/velero/commit/069c280f03b5c68810d8ef7787ceae3e710f61c2

**Comment by qiuming-best** [comment]
What you mentioned was a bug that was not fixed in v1.12.1. the https://github.com/vmware-tanzu/velero/commit/069c280f03b5c68810d8ef7787ceae3e710f61c2 is merged into the main branch only.

if you use the `velero:velero:main` image it will be all that you expected.

 
<img width="1077" alt="image" src="https://github.com/vmware-tanzu/velero/assets/8985337/a853ccbe-9cf6-45fb-b6a6-a4bb46d1a270">

**Comment by HaveFun83** [comment]
@qiuming-best thanks for your reply. 
But we run velero in production environments and need stable and reliable releases. 
Can https://github.com/vmware-tanzu/velero/commit/069c280f03b5c68810d8ef7787ceae3e710f61c2 be picked for the next release?

**Comment by HaveFun83** [comment]
@qiuming-best @allenxu404 
any news here?

**Comment by allenxu404** [comment]
Given that this issue did not originate from 1.12, we recommend including the fix(#6784) in Velero 1.13 rather than cherry-picking it into the 1.12 minor version.

**Comment by allenxu404** [comment]
OK, if that's the case, I think we should cherry-pick it to 1.12.2.

@qiuming-best Can you help to add label `target/1.12.2` to this issue?  I will cherry-pick the fix when we release 1.12.2.
